substrate/frame/bounties/src/lib.rs

```diff
@@ -845,15 +969,21 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			#[pallet::compact] bounty_id: BountyIndex,
 			curator: AccountIdLookupOf<T>,
-			#[pallet::compact] fee: BalanceOf<T, I>,
+			#[pallet::compact] fee: AssetBalanceOf<T, I>,
 		) -> DispatchResult {
 			let max_amount = T::SpendOrigin::ensure_origin(origin)?;
 			let curator = T::Lookup::lookup(curator)?;
 			Bounties::<T, I>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
 				// approve bounty
 				let bounty = maybe_bounty.as_mut().ok_or(Error::<T, I>::InvalidIndex)?;
+				let native_amount =
+					<T as pallet_treasury::Config<I>>::BalanceConverter::from_asset_balance(
+						bounty.value,
+						bounty.asset_kind.clone(),
+					)
+					.map_err(|_| pallet_treasury::Error::<T, I>::FailedToConvertBalance)?;
 				ensure!(
-					bounty.value <= max_amount,
+					native_amount <= max_amount,
```

Comment:
this will needs to be wrapped into the context. check the treasury spend call for reference